### PR TITLE
DOCS-3352 GitLab CI/CD Integration Documentation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Adds doc page for GitLab under CI/CD Integrations.

### Motivation

DOCS-3352

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/synthetics/cicd_integrations/gitlab.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  tag: "Blog"
+  text: "Run Datadog Synthetic tests in your GitLab pipelines"
+- link: "/continuous_integration/setup_pipelines/gitlab/"
+  tag: "Documentation"
+  text: "Set up Tracing on a GitLab Pipeline"
+---
+
+## Overview
+
+Run Synthetic tests in your [GitLab][1] pipelines, block deployments, and trigger rollbacks to ensure your code your code is added in production when your essential business workflows are functioning as expected.
```
Comment (hestonhoffman) [inline content/en/synthetics/cicd_integrations/gitlab.md:16]:
```suggestion
Run Synthetic tests in your [GitLab][1] pipelines, block deployments, and trigger rollbacks to ensure your code is added in production when your essential business workflows are functioning as expected.
```

**Review by hestonhoffman — APPROVED** [review]
One edit and I sent you a Slack DM to check something

File: content/en/synthetics/cicd_integrations/gitlab.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+### Test output
+
+This example demonstrates the pipeline has identified the configuration file and is running the test:
+
+{{< img src="synthetics/cicd_integrations/gitlab/synthetic_test_run.png" alt="A Synthetic test running in GitLab" style="width:100%;" >}}
+
+A successful test output returns the following in GitLab:
+
+{{< img src="synthetics/cicd_integrations/gitlab/successful_test_run.png" alt="A successful Synthetic test run result in a GitLab pipeline" style="width:100%;" >}}
```
Comment (peculiarism) [inline content/en/synthetics/cicd_integrations/gitlab.md:91]: @alai97 I wonder if we should change the images, they are great. I'll ping you on Slack

File: content/en/synthetics/cicd_integrations/_index.md [hunk 3]
```diff
@@ -44,6 +44,7 @@ To get started, see [Integrations](#integrations) and [use the API](#use-the-api
 {{< whatsnext desc="With Synthetics and CI/CD, you can run Synthetic tests in the CI platform provider of your choice:" >}}
```
Comment (peculiarism) [inline content/en/synthetics/cicd_integrations/_index.md:44]: Should we also change this to something like: 

```suggestion
{{< whatsnext desc="With Synthetics and CI/CD, you can run Synthetic tests in any CI platform provider of choice. Here are some documented integrations:" >}}
```

I'm not sure if it's great wording, but I wanted to highlight that the ability to run Synthetic tests in CI is not limited to just these providers.

File: content/en/synthetics/cicd_integrations/gitlab.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  tag: "Blog"
+  text: "Run Datadog Synthetic tests in your GitLab pipelines"
+- link: "/continuous_integration/setup_pipelines/gitlab/"
+  tag: "Documentation"
+  text: "Set up Tracing on a GitLab Pipeline"
+---
+
+## Overview
+
+Run Synthetic tests in your [GitLab][1] pipelines, block deployments, and trigger rollbacks to ensure your code is added in production when your essential business workflows are functioning as expected.
```
Comment (peculiarism) [inline content/en/synthetics/cicd_integrations/gitlab.md:16]: The hyperlink for GitLab links to the integration page which uses a lower case L (`Gitlab`), should we ask the team to update it?

File: content/en/synthetics/cicd_integrations/gitlab.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+stages: 
+  - test
+synthetic-tests:
+  stage: test
+  script: 
+    - npm install -g @datadog/datadog-ci
+    - datadog-ci synthetics run-tests --apiKey "$DATADOG_API_KEY" --appKey "$DATADOG_APP_KEY" -s ‘tag:e2e-tests’
+{{< /code-block >}}
+
+### Run tests using variable overrides
```
Comment (peculiarism) [inline content/en/synthetics/cicd_integrations/gitlab.md:55]: Let's add a small description:

```suggestion
### Run tests using variable overrides

You might have different test users or data specific to your CI/CD environments. You can override these variables using the `-v` command. Find out more in the datadog-ci NPM package documentation for the [Synthetics command](https://github.com/DataDog/datadog-ci/tree/master/src/commands/synthetics)
```

**Review by peculiarism — CHANGES_REQUESTED** [review]

File: content/en/synthetics/cicd_integrations/gitlab.md (hunk 4 above)
Comment (alai97) [reply content/en/synthetics/cicd_integrations/gitlab.md:16]: I can open a PR to update the case 👍

File: content/en/synthetics/cicd_integrations/_index.md (hunk 3 above)
Comment (alai97) [reply content/en/synthetics/cicd_integrations/_index.md:44]: How about something like this?

File: content/en/synthetics/cicd_integrations/gitlab.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+
+## Further reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: /integrations/gitlab/
+[2]: https://docs.gitlab.com/ee/ci/pipelines/
+[3]: https://www.npmjs.com/package/@datadog/datadog-ci
+[4]: /synthetics/cicd_integrations/configuration
```
Comment (alai97) [inline content/en/synthetics/cicd_integrations/gitlab.md:101]:
```suggestion
[4]: /synthetics/cicd_integrations/configuration
[5]: https://github.com/DataDog/datadog-ci/tree/master/src/commands/synthetics
```

**Review by peculiarism — APPROVED** [review]

**Pull request merged**